bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageImpl.java

```diff
@@ -196,4 +230,255 @@ public boolean hasCloudconfigSupport() {
         }
         return hasCloudconfigSupport;
     }
+
+    @Nullable
+    @Override
+    public String getExportedHierarchyType() {
+        return HierarchyTypes.PAGE;
+    }
+
+    @Nonnull
+    @Override
+    public Map<String, ? extends Page> getExportedChildren() {
+        if (childPages == null) {
+            childPages = getChildPageModels(request, Page.class);
+        }
+
+        return childPages;
+    }
+
+    @Nonnull
+    @Override
+    public String getExportedPath() {
+        return currentPage.getPath();
+    }
+
+    @Nullable
+    @Override
+    public String getRootUrl() {
+        if (currentStyle != null && currentStyle.containsKey(PR_IS_ROOT)) {
+            return getModelUrl(request, currentPage);
+        }
+
+        com.day.cq.wcm.api.Page page = getRootPage();
```

Comment:
why not do the import of `com.day.cq.wcm.api.Page`?

Reply:
name conflict with already imported `com.adobe.cq.wcm.core.components.models.Page`